### PR TITLE
task 5

### PR DESCRIPTION
**Pull request by leahstone7**
…ption for task 5

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -116,4 +116,43 @@ describe("GET /api/articles error handling", () => {
...
+      comments.forEach((comment) => {
+        expect(comment).toHaveProperty("comment_id")
+        expect(comment).toHaveProperty("votes")
+        expect(comment).toHaveProperty("created_at")
+        expect(comment).toHaveProperty("author")
+        expect(comment).toHaveProperty("body")
+        expect(comment).toHaveProperty("article_id")
+      })
+    })
+  })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:138]: Great that you've made this test flexible 
Don't forget to add a length check to this test - or you could have a false positive if the endpoint returns an empty array 🔍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -116,4 +116,43 @@ describe("GET /api/articles error handling", () => {
...
+    .then((response) => {
+      const comments = response.body.comments
+      expect(comments).toBeSortedBy("created_at", {descending: true})
+      comments.forEach((comment) => {
+        expect(comment).toHaveProperty("comment_id")
+        expect(comment).toHaveProperty("votes")
+        expect(comment).toHaveProperty("created_at")
+        expect(comment).toHaveProperty("author")
+        expect(comment).toHaveProperty("body")
+        expect(comment).toHaveProperty("article_id")
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:135]: You could also check that each comment has the correct `article_id`, i.e. 3 in this example

File: __tests__/app.test.js [hunk 3]
```diff
@@ -116,4 +116,43 @@ describe("GET /api/articles error handling", () => {
...
+    })
+  })
+  test("400: responds with error if article_id is invalid", () => {
+    return request(app)
+    .get("/api/articles/notAnID/comments")
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe("Bad request")
+    })
+  })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:157]: Nice error handling tests 👍

File: app/models/article_id_comments.model.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const db = require("../../db/connection")
+
+exports.selectAllArticleComments = (articleId) => {
+    return db
+    .query(`SELECT comment_id, votes, created_at, author, body, article_id FROM comments WHERE article_id = $1
+    ORDER BY created_at DESC`, [articleId])
+    .then((result) => {
+        console.log(result.rows)
```
Comment (nc-katherine-hurst) [inline app/models/article_id_comments.model.js:8]: Don't forget to remove console logs from finalised code 🔍

File: app/controllers/article_id_comments.controller.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const { selectAllArticleComments } = require("../models/article_id_comments.model.js")
+
+exports.getArticleComments = (req, res, next) => {
+    const { article_id } = req.params
+
+    if(isNaN(article_id)){
```
Comment (nc-katherine-hurst) [inline app/controllers/article_id_comments.controller.js:6]: This is working using a custom error, but you could also consider using some error handling middleware that catches the resulting PSQL error, this would mean that you could catch this error over multiple endpoints without having to add a custom error to check the typing each time 🔍 
Here's some relevant notes if you'd like to try this!
l2c.northcoders.com/courses/sd-notes/back-end/#sectionId=error-handling,step=postgresql-errors

File: endpoints.json [hunk 6]
```diff
@@ -40,5 +40,29 @@
...
+        {
+          "comment_id": 10,
+          "votes": 0,
+          "created_at": "2020-06-20T07:24:00.000Z",
+          "author": "icellusedkars",
+          "body": "git push origin master",
+          "article_id": 3
+        }
+      ]
+    }
```
Comment (nc-katherine-hurst) [inline endpoints.json:65]: Great!

**Review by nc-katherine-hurst — COMMENTED** [review]
Great work, please find your feedback below 🙂

File: __tests__/app.test.js [hunk 7]
```diff
@@ -116,4 +116,43 @@ describe("GET /api/articles error handling", () => {
...
 })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:158]: Sorry forgot one more thing 🧠  - it would be good to check that if an article exists but has no comments (for example article 13), you get an empty array of comments, and a status 200.

**Review by nc-katherine-hurst — COMMENTED** [review]
Sorry forgot one more thing!
